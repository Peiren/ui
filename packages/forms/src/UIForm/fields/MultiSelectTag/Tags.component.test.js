import React from 'react';
import { shallow, mount } from 'enzyme';
import Tags from './Tags.component';

describe('MultiSelectTag field > Tags', () => {
	const props = {
		onRemoveTag: jest.fn(),
		readonly: false,
		titleMap: [{ name: 'toto', value: 'titi' }, { name: 'tata', value: 'tutu' }],
		value: ['titi', 'lol', 'tutu'],
	};

	it('should display tags list', () => {
		// when
		const wrapper = shallow(<Tags {...props} />);

		// then
		expect(wrapper.node).toMatchSnapshot();
	});

	it('should display readOnly tags list', () => {
		// given
		const roProps = {
			...props,
			readonly: true,
		};

		// when
		const wrapper = shallow(<Tags {...roProps} />);

		// then
		expect(wrapper.node).toMatchSnapshot();
	});

	it('should call onTagsMount', () => {
		// given
		const onTagsMount = jest.fn();
		const mountProps = {
			...props,
			onTagsMount,
		};

		// when
		mount(<Tags {...mountProps} />);

		// then
		expect(onTagsMount).toBeCalled();
	});

	it('should call onRemoveTag', () => {
		// given
		const onRemoveTag = jest.fn();
		const removeProps = {
			...props,
			onRemoveTag,
		};
		const wrapper = mount(<Tags {...removeProps} />);

		// when
		wrapper.find('button.tc-badge-delete-icon').at(1).simulate('click');

		// then
		expect(onRemoveTag).toBeCalledWith(expect.anything(), 1);
	});
});
