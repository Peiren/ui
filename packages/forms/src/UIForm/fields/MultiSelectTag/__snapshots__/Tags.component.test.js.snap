// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MultiSelectTag field > Tags should display readOnly tags list 1`] = `
<div
  className="tags"
>
  <Translate(Badge)
    label="toto"
  />
  <Translate(Badge)
    label="lol"
  />
  <Translate(Badge)
    label="tata"
  />
</div>
`;

exports[`MultiSelectTag field > Tags should display tags list 1`] = `
<div
  className="tags"
>
  <Translate(Badge)
    label="toto"
    onDelete={[Function]}
  />
  <Translate(Badge)
    label="lol"
    onDelete={[Function]}
  />
  <Translate(Badge)
    label="tata"
    onDelete={[Function]}
  />
</div>
`;
